__test__/port.test.js

```diff
@@ -0,0 +1,42 @@
+const Port = require('../src/port.js')
+
+describe('port constructor', () => {
+    describe('has ports and ships', () => {
+        let port
+        let theBlackPearl
+        let theFlyingDutchman
+        
+        beforeEach(() => {
+            port = new Port ('Amsterdam')
+            ship = jest.fn()
+            theBlackPearl = jest.fn()
+            theFlyingDutchman = jest.fn()
+        })
+        it('returns an object', () => {
+            
+            expect(new Port()).toBeInstanceOf(Object)
+        })
+        it('has a name property', () => {
+    
+            expect(port.name).toBe('Amsterdam')
+        })
+        it('can add a ship', () => {
+    
+            port.addShip(ship)
+    
+            expect(port.ships).toContain(ship)
+        })
+        it('can remove a ship', () => {
+
+            const port = new Port('Amsterdam')
+            const theBlackPearl = {}
+            const theFlyingDutchman = {}
+
+            port.addShip(theBlackPearl)
+            port.addShip(theFlyingDutchman)
+            port.removeShip(theBlackPearl)
+    
+            expect(port.ships).toEqual([theBlackPearl])
```

Comment:
You create `theBlackPeral` and `theFlyingDutchman` variables in the `beforeEach` but you only make use of them in the last test, where you create them again anyway. if you only need a variable for one test only create it for that test, i.e. remove them from the `beforeEach` and remove `let theBlackPearal` and `let theFlyingDutchman`

Reply:
Same as ship, as it is only used in the `'can add a ship'` test. Also, I'm surprised the test works, as you didn't initialise a variable for ship beforehand, but after some testing that doesn't seem to matter in these tests and I'm not quite sure why. I tested this in some projects I made and they fail if I don't so I will have to look into that some more to see if I can figure out why